src/components/FormFillnaStep.vue

```diff
@@ -0,0 +1,142 @@
+    </div>
+    <WidgetAutocomplete
+      id="columnInput"
+      v-model="step.column"
+      name="Fill null values in:"
+      :options="columnNames"
+      @input="setSelectedColumns({ column: step.column })"
+      placeholder="Enter a column"
+    ></WidgetAutocomplete>
+    <WidgetInputText id="valueInput" v-model="step.value" name="With:" placeholder="Enter a value"></WidgetInputText>
+    <div class="widget-form-action">
+      <button
+        class="widget-form-action__button widget-form-action__button--validate"
+        @click="validateStep"
+      >OK</button>
+      <button
+        class="widget-form-action__button widget-form-action__button--cancel"
+        @click="cancelEdition"
+      >Cancel</button>
+    </div>
+    <div v-if="errors" class="errors">
+      <ul>
+        <li v-for="(error, index) in errors" :key="index">{{ error.dataPath }}: {{ error.message }}</li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import _ from 'lodash';
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator';
+import FormMixin from '@/mixins/FormMixin.vue';
+import { Pipeline } from '@/lib/steps';
+import fillnaSchema from '@/assets/schemas/fillna-step__schema.json';
+import WidgetInputText from './WidgetInputText.vue';
+import WidgetAutocomplete from '@/components/WidgetAutocomplete.vue';
+import { Getter, Mutation, State } from 'vuex-class';
+import { StepFormComponent } from '@/components/formlib';
+
+interface FillnaStepConf {
+  column: string;
+  value: string;
+}
+
+@StepFormComponent({
+  vqbstep: 'fillna',
+  name: 'form-fillna-step',
+  components: {
+    WidgetAutocomplete,
+    WidgetInputText,
+  },
+})
+export default class FillnaStepForm extends Mixins(FormMixin) {
+  @Prop({
+    type: Object,
+    default: () => ({
+      column: '',
+      value: '',
+    }),
+  })
+  initialValue!: FillnaStepConf;
+
+  @Prop({
+    type: Boolean,
+    default: true,
+  })
+  isStepCreation!: boolean;
+
+  step: FillnaStepConf = { ...this.initialValue };
+
+  @State pipeline!: Pipeline;
+  @State selectedStepIndex!: number;
+
+  @Mutation selectStep!: (payload: { index: number }) => void;
+  @Mutation setSelectedColumns!: (payload: { column: string }) => void;
+
+  @Getter selectedColumns!: string[];
+  @Getter columnNames!: string[];
+  @Getter computedActiveStepIndex!: number;
+
+  @Watch('selectedColumns')
+  onSelectedColumnsChanged(val: string[], oldVal: string[]) {
+    if (!_.isEqual(val, oldVal)) {
+      this.step.column = val[0];
+    }
+  }
+
+  created() {
+    this.schema = fillnaSchema;
+    this.setSelectedColumns({ column: this.initialValue.column });
+  }
+
+  validateStep() {
+    const ret = this.validator(this.step);
+    if (ret === false) {
+      this.errors = this.validator.errors;
+    } else {
+      this.errors = null;
+      this.$emit('formSaved', { name: 'fillna', ...this.step });
+    }
+  }
+
+  cancelEdition() {
```

Comment:
Isn't it the time to move this logic into `FormMixin` ?

Reply:
You're right! I suggest we do that in another PR, though.